### PR TITLE
Updates asciidoctor version used in Gradle build to version 4.0.0-alpha.1 

### PR DESCRIPTION
**Pull request by tresat**
This will allow the Gradle build to benefit from configuration cache compatibility.

This leaves the version used in smokes tests at 3.2 until that upgrade is ready (prepared in [this PR](https://github.com/gradle/gradle/pull/23427)).

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -1064,11 +1065,21 @@
             <sha256 value="05621bf96e33332f1fddb3522cd17d3a1afebefbef5d123b897f84ece0667fc6" origin="Verified" reason="Artifact is not signed"/>
          </artifact>
       </component>
+      <component group="org.asciidoctor" name="asciidoctor-gradle-base" version="4.0.0-alpha.1">
+         <artifact name="asciidoctor-gradle-base-4.0.0-alpha.1.jar">
+            <sha256 value="59281a4861a380148bfb9924abcb1287bccdb6484955ac447577cf8bde17eefc" origin="Generated by Gradle"/>
```
Comment (eskatos) [inline gradle/verification-metadata.xml:1070]: :x: The `origin` should be updated, even better would be adding a `reason`. See the comments at the beginning of this file.

**Review by eskatos — CHANGES_REQUESTED** [review]
I'm going to try locally to see if we can declare more tasks compatible with CC on our build

**Comment by eskatos** [comment]
It doesn't help much yet, see
* https://github.com/asciidoctor/asciidoctor-gradle-plugin/issues/564#issuecomment-1380450159

I guess we should then wait for another version

**Comment by big-guy** [comment]
@tresat We should take this out of 8.1

**Comment by ov7a** [comment]
Closing this as obsolete. See
* #27839

**Pull request closed without merging**